test/hotspot/gtest/gc/shared/test_preservedMarks.cpp

```diff
@@ -29,7 +29,10 @@
 // Class to create a "fake" oop with a mark that will
 // return true for calls to must_be_preserved().
 class FakeOop {
-  oopDesc _oop;
+  // Align at least to 8 bytes, otherwise the lowest address bit
+  // could be interpreted as 'self-forwarded' when encoded as
+  // forwardee in the mark-word.
+  alignas(BytesPerLong) oopDesc _oop;
```

Comment:
Could you explain why this is required? (I'd expect `oopDesc` to be properly aligned already, since it's >= 8 bytes.)

Reply:
This is a problem with x86_32 builds, where oopDesc is only 4 byte. It is only a problem in this test, in real world, objects in the heap are always at least 8-byte-aligned.

Reply:
Even on 32bits system, shouldn't `_mark` = 4 bytes and `_metadata` = 4 bytes, so the total size of `oopDesc` is 8 bytes?

Reply:
The alignment of a struct must just ensure that the alignments requirement of its fields are satisfied (in the general case) So a struct of two 4 byte ints will have an `alignof` 4 bytes and a `sizeof` 8 bytes
```c++
#include <cstdint>
struct S {
    uint32_t a;
    uint32_t b;
};

static_assert(alignof(S) == 4);
static_assert(sizeof(S) == 8);
```

Reply:
Thank you for the explanation; I mistakenly thought a struct is aligned to its size...